pennylane/optimize/adaptive.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Adaptive optimizer"""
+# pylint: disable= no-value-for-parameter
+import pennylane as qml
+
+from pennylane import numpy as np
+
+
+@qml.qfunc_transform
+def append_gate(tape, params, gates):
+    """Append parameterized gates to an existing tape.
+
+    Args:
+        tape (QuantumTape): quantum tape to transform by adding gates
+        params (array[float]): parameters of the gates to be added
+        gates (list[Operator]): list of the gates to be added
+    """
+    for o in tape.operations:
+        qml.apply(o)
+
+    for i, g in enumerate(gates):
+        g.data[0] = params[i]
+        qml.apply(g)
+
+    for m in tape.measurements:
+        qml.apply(m)
+
+
+class AdaptiveOptimizer:
+    r"""Adaptive optimizer.
```

Comment:
@soranjh something to keep in mind is that the first sentence of a docstring is what is displayed in summary tables :)

![image](https://user-images.githubusercontent.com/2959003/198046991-2ce9eefb-261a-46d9-9def-502a66277c15.png)

So worth putting something more descriptive (but still relatively concise) here.